src/deadline_test_fixtures/deadline/worker.py

```diff
@@ -223,10 +225,60 @@ def start(self) -> None:
         self._start_worker_agent()
 
     def stop(self) -> None:
+        worker_id = self.worker_id
         LOG.info(f"Terminating EC2 instance {self.instance_id}")
         self.ec2_client.terminate_instances(InstanceIds=[self.instance_id])
+
+        """
+        When using autoscaling workers are deleted from fleets, 
+        for non autoscaling fleets we need to manually delete the worker.
+        """
+        if not self.configuration.fleet.autoscaling:
+            self.delete_worker_from_fleet(worker_id)
+
         self.instance_id = None
 
+    def delete_worker_from_fleet(self, worker_id: str):
```

Comment:
Should we simplify the name from `delete_worker_from_fleet(...)` to just `delete(...)`?

Reply:
done.